src/v/raft/recovery_stm.cc

```diff
@@ -149,10 +149,12 @@ ss::future<> recovery_stm::do_recover(ss::io_priority_class iopc) {
         co_return;
     }
 
-    co_await replicate(
-      std::move(*reader),
-      should_flush(follower_committed_match_index),
-      std::move(read_memory_units));
+    auto flush = should_flush(follower_committed_match_index);
+    if (flush) {
```

Comment:
use explicit bool test. otherwise the reviewer has to go figure out that this is not a bool but an integer and you are relying on implicit conversion of 0=false.

Reply:
Good point, updating the commit to make the condition explicit.